### ISSUE TITLE
Spider-Man: Water rendering is broken

### ISSUE THREAD

**Issue by K0bin** [HansKristian-Work/vkd3d-proton#1191]
Water rendering is broken.

![20220813172724_1](https://user-images.githubusercontent.com/1131720/184500726-41943c7d-4f8f-4532-9543-a49056d72274.jpg)
![20220813172844_1](https://user-images.githubusercontent.com/1131720/184500736-54598bfe-71a2-4d6a-b8b0-572b745abc5c.jpg)

Looks like another issue with a depth prepass, unfortunately I couldn't run the game with Renderdoc.

Nvidia RTX 3090, 515.65.01
Proton Experimental (as of the 13th of August), shipping d00d035321c9f817e058985e32fe6876ba746de6

**Comment by K0bin** [comment]
Unfortunately this isn't 100% reproducible. I've had it happen multiple times but it doesn't happen every time I start the game. Might be related to the time of day.

**Comment by K0bin** [comment]
It happens at dusk. It's fine after initially loading into the game. Fast travelling triggers it.

Renderdoc works with the game after all (just needs no_upload_hvv to be usable) but unfortunately the bug doesn't occur with Renderdoc attached. .__.

**Comment by FSKiller** [comment]
This also happened to me with a RTX 2070 with Driver version: 515.43.04

**Comment by HansKristian-Work** [comment]
So this is an NV exclusive bug?

**Comment by K0bin** [comment]
I think it is NV exclusive and the bug itself is not impacted by no_upload_hvv.

I am however not sure whether this still happens after the latest patch.

**Comment by halfmexican** [comment]
Nope this still occurs

**Comment by Jaxseven** [comment]
I've been testing this issue in this location during the day and it's not 100% reproducible even in the same time of day or location. I've managed to find out that having the launch option RADV_PERFTEST=sam and SAM enabled on my bios seems to push back the black reflection further to the point of being barely noticeable during the day time. Have not tested any other time of day and lighting is still odd. 
![20221029104711_1](https://user-images.githubusercontent.com/47583647/198838554-4a6b1647-e85d-488b-bcb7-481c44612273.jpg)

**Comment by K0bin** [comment]
Wait, you can reproduce it on AMD GPUs?

**Comment by Jaxseven** [comment]
> 

Yeah sorry, I guess I just jumped in. I'm running Nobara on my 6700 XT/5900x system. I didn't experience this before on my Manjaro so I'm try to track down the cause and got linked here.

**Comment by sriharshachilakapati** [comment]
I've been facing this glitch in rendering water sometimes (rare, like 1 in 10 runs). This happens when I just finish a mission and go to the shore to see ocean. Just a quit to main menu and go back in would fix it though.

Using High Preset in game, running at 1080p with DLSS. Screen resolution (in System Settings) is 4k. Using Proton GE 7-38 with patched vkd3d-2.7-proton.

Kernel is 6.0.5-1-MANJARO. NVIDIA Driver version is 520.56.06.

CPU is 11th Gen Intel i5-11400H (12) @ 4.500 GHz GPU is NVIDIA GeForce RTX 3060 Mobile / Max-Q RAM is 16GB DDR4 Running two SSDs (Btrfs) with OS on one SSD and Home partition on secondary one Both are NVME M.2 SSDs (500GB and 1TB respectively)

Running Manjaro (testing branch) with all packages up-to-date.

CPU goes arond 60-70°C while GPU goes to around 90°C.

Btw, not just in Hell's Kitchen, it happens on all shores for me. Will see if fast travel will affect this.

**Comment by Trevonn** [comment]
Also happens in Miles Morales

RX 6600M - Mesa-git
VKD3D-Proton-git
Linux 6.0.9

The behaviour does change when you add `VKD3D_CONFIG=no_upload_hvv` to the launch options but the issue is still present

[Images](https://imgur.com/a/BChej7Z)

Images 2 and 3 have `VKD3D_CONFIG=no_upload_hvv` applied

**Comment by sriharshachilakapati** [comment]
What does no_upload_hvv do?

**Comment by K0bin** [comment]
It prevents VKD3D-Proton from putting D3D12_HEAP_UPLOAD buffers into CPU visible VRAM (resizable bar).
